components/support/error/macros/src/signature.rs

```diff
@@ -0,0 +1,35 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+use syn::{spanned::Spanned, GenericArgument, PathArguments};
+
+const WRONG_RETURN_TYPE_MSG: &str = "Expected Result<T> or Result<T, E>";
+
+pub(crate) fn validate(sig: &syn::Signature) -> syn::Result<()> {
```

Comment:
I think this is checking if the return type has 1 generic argument -- is this right?  I think this should be documented.

Or maybe it should be removed.  I'm not sure what it's guarding against and I'm slightly worried that type aliases could cause errors.

Reply:
That's a good point, I think we can do without this check and leave it to the Rust compiler to error if the return types are incorrect!